enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -241,6 +247,46 @@ struct CacheAnalysis {
     return seen[obj] = mustcache;
   }
 
+  // recursive floodfill-like function
+  // propagates enzyme_nocache from the arguments to specific values
+  bool compute_nocache(Value* obj, int depth = 0)
+  {
+    if (auto s = nocache_seen.find(obj); s != nocache_seen.end()) {
+      return s->second;
+    }
+
+    if (++depth > 5)
+        return false;
+
+    if (auto arg = dyn_cast<Argument>(obj))
+    {
+      return nocache_seen[obj] = !overwritten_args[arg->getArgNo()];
+    }
+    else if (auto pn = dyn_cast<PHINode>(obj))
+    {
+      for (auto &val : pn->incoming_values())
```

Comment:
This wouldn't work for recursive phi nodes, if that's desired. 

E.g. phi = [ nocachearg if entry] else [ phi+1 if not entry ].

We could set up a map of values to states,  initialized to unknown and build something to try to support this.